### ISSUE TITLE
MatCat not working anymore

### ISSUE THREAD

**Issue by CyrilleB79** [NSoiffer/MathCAT#309]
MatCat is not working anymore. I do not know since when.

* Current NVDA version 2024.4.1rc1
* Current MatCat version: 0.6.6

Log trying to read the quadratic equation Wikipedia page.:
[nvda.log](https://github.com/user-attachments/files/17737360/nvda.log)

**Comment by NSoiffer** [comment]
From your log:
```
MathCAT 0.6.6 installed. Using rules dir: C:\Users\Cyrille\AppData\Local\Temp\nvdaMatCat\addons\MathCAT\globalPlugins\MathCAT\Rules
ERROR - external:globalPlugins.MathCAT.MathCAT.MathCAT.__init__ (19:14:00.863) - MainThread (10448):
Couldn't open zip file \\?\C:\Users\Cyrille\AppData\Local\Temp\nvdaMatCat\addons\MathCAT\globalPlugins\MathCAT\Rules\Languages\en\fr.zip: Le fichier spécifié est introuvable. (os error 2).
```

I see a few things that seem different:
1. On the machines I've seen, NVDA is in `...AppData\Roaming\nvda`. This implies it is in `Local`, not `Roaming`. That's not necessarily wrong, but it is different.
2. This part implies that you are definitely doing something different: `...Temp\nvdaMatCat\...`.  Does `\Users\Cyrille\AppData\Local\Temp\nvdaMatCat\addons\MathCAT\globalPlugins\MathCAT` exist?
3. It appears you are trying to read in French, which MathCAT v0.6.6 doesn't support.  Starting sometime in May or June, MathCAT started zipping up the various language files to save space (5mb and counting). If it is unzipped, it should still work, but if it is zipped, perhaps there is a problem with that. What are the contents of `...Rules\Languages\fr` (assuming there such a directory)?

Have you made progress on a French translation? If so, do you want to contribute it back to the main repo so others can benefit from it? Also so that it incorporated into tests before a release.

**Comment by CyrilleB79** [comment]
Sorry I have forgotten to provide you some information:
* I get such errors in my normal configuration
* to isolate the issue, I have run NVDA with the following command line:
  `nvda -c %tmp%\nvdaMatCat --lang=fr`
  The `-c` allows me to start from a clean new config. I have only modified a few non significant config parameters (see the config at the beginning of the log), installed MatCAT (it's the only installed add-on) and restarted.
  The `--lang` parameter allows me to run something that you can also run on your side; if I do not use it, NVDA will use the default system language, which is French for me, but probably something else for you.

To summarize, the issue also occurs when using in my normal config in ""C:\Users\Cyrille\AppData\Roaming\nvda" and when using no `--lang` parameter, with my French Windows.

I do not expect MatCAT to read math in French, but at least in English out of the box.

Regarding the French translation, I have no time to do it for now since it's not one of my priorities; I'll let you know if/when I turn back to it. Have you received other offers for French translations?

**Comment by NSoiffer** [comment]
The bad news is that I can't replicate the problem, even using your command. I upgraded to 2024.4.1 from 2024.4 and still I don't have any problems. At some point, NVDA stopped working on reading any web pages, so I did a reboot. There was a Windows update pending and I've found that in the past, in the pending state, I often have strange problems. But after the update, NVDA worked fine and there were no error messages from MathCAT. It's a long shot, but can you try rebooting?

**Comment by andrew-l-d** [comment]
I am not having any problems here with NVDA 2024.4.1 and latest MathCAT.

**Comment by NSoiffer** [comment]
@CyrilleB79: did you try a reboot?

**Comment by CyrilleB79** [comment]
Yes, I did a reboot (even many ones).

When I have time, I may try to investigate, but for now, it's low priority for me.

**Comment by NSoiffer** [comment]
I'm going to close it for now since neither I nor @andrew-l-d can replicate it. Please re-open if you get more information.

Note that the current 2025.1 alpha builds have broken navigation in MathCAT (it is somewhere in NVDA's braille handling code, not MathCAT... or maybe the interface for dealing with braille changed. I still need to file a bug report.
